### ISSUE TITLE
Feature Request: Filter by Drive (easy pause torrents with data on specific drives)

### ISSUE THREAD

**Issue by enken** [transmission/transmission#858]
I often wish I could pause all torrents downloading to/seeding from a specific external drive (or all externals) so that I can disconnect it without interrupting other torrents. I'm using Transmission on a macOS laptop: Docking at a desk with USB drives vs. moving laptop to another area.

Being able to filter torrents by data location would allow selection of torrents by drive. "Select All" and "Pause Selected" and "Resume Selected" would allow for quick pause/disconnect (and connect/resume) without interruption, if one could filter by drive.

This could also help after the issue where "data location missing" during Transmission start, or when a drive is disconnected unintentionally (such as power loss). (Here I also wonder if an option to resume automatically upon drive re-connection is possible.)

[#822](https://github.com/transmission/transmission/pull/822) is merged and offers a potential work-around of manually labeling torrents. Filtering by data location or drive still seems to me like a worthy idea on its own.

**Comment by ckerr** [comment]
...maybe? This might be doable, it's more a question of making sure that it doesn't clutter the UI.

If, for example, the pre-existing 'search' in the  main window looked not just for substrings of downloaded files but also their download foldernames, that might be a simple way of doing this without making the UI more complicated.

**Comment by enken** [comment]
If searching for “/Volumes/ExtDisk/folder name” works, then that gets it done! and opens up some possibilities for finding everything in particular folders even.

I’m brainstorming for my MacOS client: if labels become a thing, then it would be useful if the Preferences had a section for dynamically applied “Smart Labels”, where one use is torrents pointing to particular folder or drive could have a label applied (/removed) automatically.  (This would help rather than typing out a search each time.)

Smart Labels could also be useful for other rule-based applications, and offer a great shortcut to repeated searches.

**Comment by GScottElshaw** [comment]
> Smart Labels

Does the 'Groups' function not help to sort by device?

**Comment by enken** [comment]
since Groups are flat and exclusive I use them to sort transfers by content type, or relevancy to current projects, or certain trackers (so I don’t have to run a search).  If I used Groups for drive/folder, then I would either have very few Groups and a jumbled mess list of transfers, or repeated Groups with the drive/folder difference and be unable to see a complete view by drive/folder quickly.

Labels should make everything better, though it’s helpful to not have to manually add every label every time.

Smart Labels is, I guess, a wishful idea akin to the “Assign Group … based on criteria option” that currently only looks inside a torrent and is applied once, but I think would be usefully expanded to include other transfer details and would especially help workflows to not have to repeat search criteria.

I conceive of clicking the Pin icon, selecting a label (like Group can be filtered)

Having a history of frequently used searches (and more powerful search) is a more realistic wish.

**Comment by GScottElshaw** [comment]
The ability to select and have metadata appeals to me - not everything needs to be visible.

**Comment by Coeur** [comment]
Copy-pasting my comment from #1817:
>I see two possible implementation approaches:
>
>A. We generate in `setFilterText:` some new `FileListNode`s each time we filter, to represent the folders with less children than without filter.
>
>B. Or we also filter in each delegate method like `outlineView:numberOfChildrenOfItem:` and `outlineView:child:ofItem:`
>
>**And there may be the question of selecting/unselecting a folder which is in filtered mode: does that select/unselect only the filtered items, or all the real children of the folder?**

**Comment by tomck** [comment]
Yes, sometimes I want to migrate all torrents stored in a directory or on a volume. It is a pain having to look at the file names, search them and move data one by one, especially with that blocking issue. The simplest way would be another sort option, sort by path. This would allow scrolling to the first file in a path such as /Volume/Data/One and select all items in that path without too much hassle.

You can sort transfers by queue order, activity, date added, even tracker, but not path.
